controllers/nonadmin_controller.go

```diff
@@ -0,0 +1,192 @@
+package controllers
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	"golang.org/x/exp/maps"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	k8serror "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+
+	oadpv1alpha1 "github.com/openshift/oadp-operator/api/v1alpha1"
+	"github.com/openshift/oadp-operator/pkg/common"
+)
+
+const (
+	nonAdminPrefix            = "non-admin-"
+	controllerManager         = "controller-manager"
+	containerName             = nonAdminPrefix + "manager"
+	nonAdminControllerManager = common.OADPOperatorPrefix + nonAdminPrefix + controllerManager
+	controlPlaneKey           = "control-plane"
+)
+
+var (
+	controlPlaneLabel = map[string]string{
+		controlPlaneKey: nonAdminPrefix + controllerManager,
+	}
+	deploymentLabels = map[string]string{
+		"app.kubernetes.io/component":  "manager",
+		"app.kubernetes.io/created-by": common.OADPOperator,
+		"app.kubernetes.io/instance":   nonAdminPrefix + controllerManager,
```

Comment:
The instance name can be `non-admin-controller`, `manager` not needed here as the `component` label signifies that.

Reply:
here I also tried to change little, by just adding prefix https://github.com/migtools/oadp-non-admin/pull/16/files#diff-ce0d597a9e5f20a8fe61c3ba5e6185a6b90cfef2c00bb5b5ef3e97b15b79f480R22

does it make sense?

Reply:
maybe just drop the `manager` ?

Reply:
as discussed with team, will make name shorter for avoiding limit name issues

Reply:
done